paddle/fluid/memory/detail/buddy_allocator.cc

```diff
@@ -243,26 +243,26 @@ BuddyAllocator::PoolSet::iterator BuddyAllocator::FindExistChunk(size_t size) {
 void* BuddyAllocator::SplitToAlloc(BuddyAllocator::PoolSet::iterator it,
                                    size_t size) {
   auto block = static_cast<MemoryBlock*>(std::get<2>(*it));
+  auto desc = cache_.LoadDesc(block);
   pool_.erase(it);
 
-  VLOG(10) << "Split block (" << block << ", " << block->total_size(cache_)
-           << ") into";
-  block->split(&cache_, size);
+  VLOG(10) << "Split block (" << block << ", " << desc->total_size << ") into";
+  block->Split(&cache_, size);
 
-  VLOG(10) << "Left block (" << block << ", " << block->total_size(cache_)
-           << ")";
-  block->set_type(&cache_, MemoryBlock::ARENA_CHUNK);
+  VLOG(10) << "Left block (" << block << ", " << desc->total_size << ")";
+  desc->type = MemoryBlock::ARENA_CHUNK;
```

Comment:
actually UpdateGuards is like a "soft version" of Save function here. If I define a SetType and put UpdateGuards inside, then I also need to define SetSize(), SetTotalSize(), SetLeftBuddy() and so on, but actually multiple actions only need to call UpdateGuards once, this is the reason that I just call UpdateGuards() explicitly

Reply:
You can define `SetType()` only for this PR. I think that `UpdateGuards` should be a private method, which hides the implementation of memory modules.

Reply:
In fact, maybe it is better to use getter method for `index`, `size`, `total_size`, `left_buddy` instead of using data members directly, which may be a risk if someone forgets to call `UpdateGuards` after setting some fields.

Reply:
done